cmd/mdatagen/internal/metadata.go

```diff
@@ -299,7 +299,8 @@ type Attribute struct {
 	// Enum can optionally describe the set of values to which the attribute can belong.
 	Enum []string `mapstructure:"enum"`
 	// Type is an attribute type.
-	Type ValueType `mapstructure:"type"`
+	Type        ValueType `mapstructure:"type"`
+	TemplateKey bool      `mapstructure:"template_key"`
```

Comment:
Now that I'm thinking about it a bit more, what about `is_template` since it will be a boolean?